students/Evan Dzurick/Part4DesignChallenge/Part4DesignChallenge.ino

```diff
@@ -0,0 +1,177 @@
+const int BLED=9; //Blue LED on Pin 9
+const int GLED=10; //Green LED on Pin 10
+const int RLED=11; //Red LED on Pin 11
+const int BUTTON=2; //The Button is connected to pin 2
+int previousMillis = 0;
+int ledState = 0;
+
+boolean lastButton = LOW; //Last Button State
+boolean currentButton = LOW; //Current Button State
+int ledMode = 0; //Cycle between LED states
+
+void setup()
+{
+   pinMode (BLED, OUTPUT); //Set Blue LED as Output
+   pinMode (GLED, OUTPUT); //Set Green LED as Output
+   pinMode (RLED, OUTPUT); //Set Red LED as Output
+   pinMode (BUTTON, INPUT); //Set button as input (not required)
+}
+
+/*
+* Debouncing Function
+* Pass it the previous button state,
+* and get back the current debounced button state.
+*/
+boolean debounce(boolean last)
+{
+   boolean current = digitalRead(BUTTON); //Read the button state
+  if (last != current) //if it's different...
+   {
+     delay(5); //wait 5ms
+     current = digitalRead(BUTTON); //read it again
+   }
+  return current; //return the current value
+}
+/*
+* LED Mode Selection
+* Pass a number for the LED state and set it accordingly.
+*/
+void setMode(int mode)
+
+
+{
+
+ // #1 LED OFF 
+ if (mode == 0)
+ {
+   digitalWrite(RLED, LOW);
+   digitalWrite(GLED, LOW);
+   digitalWrite(BLED, LOW);
+ }
+ // #2 WHITE BLINKING
+ else if (mode == 1){
+
+
+  unsigned long currentMillis = millis(); //unsigned long is integers without sign with large range of values 
+  if (currentMillis - previousMillis >= 500){
+      previousMillis = currentMillis; 
+
+    if (ledState == 0) {
+        ledState = 85;
+      } else {
+        ledState = 0;
+      }
+
+
+  analogWrite(RLED, ledState);
+  analogWrite(GLED, ledState);
+  analogWrite(BLED, ledState);
+  }
+ }
+
+ // #3 PURPLE FADES BRIGHT to DIM -> REPEAT
+ if (mode == 2){
+  unsigned long currentMillis = millis();
+  if (currentMillis - previousMillis <= 3000){
+
+    int brightness = map(currentMillis - previousMillis - 3000, 0, 5000, 255, 0);
+      analogWrite(RLED, brightness);
+      analogWrite(GLED, 0);
+      analogWrite(BLED, brightness);
+  }
+
+  else if(currentMillis - previousMillis <= 6000){
+    int brightness = map(currentMillis - previousMillis - 3000,0,5000,0,255);
+
+      analogWrite(RLED, brightness);
+      analogWrite(GLED, 0);
+      analogWrite(BLED, brightness);
```

Comment:
Level up your code: these lines are exactly the same in either `if` clause - so you could have them just once outside the conditional.